packages/core/src/util/renderer.ts

```diff
@@ -161,11 +165,32 @@ export const createDefaultValue = (schema: JsonSchema) => {
       return [];
     case 'null':
       return null;
+    case 'object':
+      return extractDefaults(schema);
     default:
       return {};
   }
 };
 
+/**
+ * Returns the default value defined in the given schema.
+ * @param schema the schema for which to create a default value.
+ * @returns {any}
+ */
+export const extractDefaults = (schema: JsonSchema) => {
+  if (schema.type === 'object' && schema.default === undefined) {
+    const result: { [key: string]: any } = {};
+    for (const key in schema.properties) {
+      const property = schema.properties[key];
```

Comment:
For each property we need to check whether it is a reference, i.e. `'$ref' in schema` and resolve accordingly